attribute-service-impl/src/main/java/org/hypertrace/core/attribute/service/model/AttributeMetadataModel.java

```diff
@@ -54,6 +65,10 @@
   private List<AttributeSource> sources = Collections.emptyList();
   private Map<String, Map<String, String>> metadata = Collections.emptyMap();
 
+  @JsonSerialize(using = ProtobufMessageSerializer.class)
+  @JsonDeserialize(using = AttributeDefinitionDeserializer.class)
```

Comment:
It works, but I'm definitely open to better ways fo accomplishing this if anyone has ideas. It seems much more error prone to manually trying to map the object and handle the compatibility here vs piggy backing off proto's existing support.

Reply:
Never though of doing it this way. That said the serializer works for all types. The deserializer can be mad generic by using java generics I think?

Reply:
Because the message contains the schema, we can be generic with the serializer. Coming from json (deser) however, we need to at a minimum know the target type (and use reflection) or better yet the builder class. I tried playing around with the deserialization context, but it doesn't appear to have anything helpful.

Reply:
Yeah. Just tried it out. Doesn't work since the builder is constraining. :( Would have to spend some real time to see if it's doable another way.